File: src/optical_element/MATLAB/g_Furion_paraboid_focus_Mirror.m
classdef  g_Furion_paraboid_focus_Mirror < g_Furion_cyliner_parabolic_focus_Mirror%追迹phase转换成角度
  properties 
    
  end
methods
function obj = g_Furion_paraboid_focus_Mirror( beam_in,ds,di,chi,theta,surface,r2,grating)   
    obj=obj@g_Furion_cyliner_parabolic_focus_Mirror( beam_in,ds,di,chi,theta,surface,r2,grating) ;%输入参数给父类   
   
end
end

methods%设置转换为center坐标的类
    function set_center(obj, beam_in,ds,di,chi,theta,surface,r2,grating)
        obj.center =g_paraboid_focus( beam_in,ds,di,chi,theta,surface,r2,grating)  ;
    end
end
end

File: src/optical_element/g_Furion_paraboid_focus_Mirror.cpp
#include "g_Furion_paraboid_focus_Mirror.h"
#include "g_paraboid_focus.h"

using namespace Furion_NS;

G_Furion_Paraboid_Focus_Mirror::G_Furion_Paraboid_Focus_Mirror(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r2, Grating* grating)
    : G_Furion_Cyliner_Parabolic_Focus_Mirror(beam_in, ds, di, chi, theta, surface, r2, grating)
{
    G_Furion_Cyliner_Parabolic_Focus_Mirror::run(beam_in, ds, di, chi, theta, surface, r2, grating);
    cout << "G_Furion_Paraboid_Focus_Mirror 初始化" << endl;
}

G_Furion_Paraboid_Focus_Mirror::~G_Furion_Paraboid_Focus_Mirror()
{
    //delete center;
    //cout << "G_Furion_Paraboid_Focus_Mirror 析构" << endl;
}

void G_Furion_Paraboid_Focus_Mirror::run(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r2, Grating* grating)
{
    cout << "G_Furion_Paraboid_Focus_Mirror的run" << endl;
    set_center(beam_in, ds, di, chi, theta, surface, r2, grating);
}

void G_Furion_Paraboid_Focus_Mirror::set_center(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r2, Grating* grating)
{
    cout << "G_Furion_Paraboid_Focus_Mirror的set_center" << endl;
    center = new G_Paraboid_Focus(beam_in, ds, di, chi, theta, surface, r2, grating);
}